sql/core/src/test/scala/org/apache/spark/sql/InjectRuntimeFilterSuite.scala

```diff
@@ -264,28 +264,25 @@ class InjectRuntimeFilterSuite extends QueryTest with SQLTestUtils with SharedSp
     }
   }
 
-  def getNumBloomFilters(plan: LogicalPlan): Integer = {
-    val numBloomFilterAggs = plan.collect {
-      case Filter(condition, _) => condition.collect {
-        case subquery: org.apache.spark.sql.catalyst.expressions.ScalarSubquery
-        => subquery.plan.collect {
-          case Aggregate(_, aggregateExpressions, _) =>
-            aggregateExpressions.map {
-              case Alias(AggregateExpression(bfAgg : BloomFilterAggregate, _, _, _, _),
-              _) =>
-                assert(bfAgg.estimatedNumItemsExpression.isInstanceOf[Literal])
-                assert(bfAgg.numBitsExpression.isInstanceOf[Literal])
-                1
-            }.sum
+  // `MergeScalarSubqueries` can duplicate subqueries in the optimized plan, but the subqueries will
```

Comment:
I'm wondering if we should run `MergeScalarSubqueries` before `InjectRuntimeFilter`, so that we don't need to change this test suite. cc @peter-toth @somani

Reply:
But, isn't it beneficial that bloom filter subqueries can be merged? Here was an example: https://github.com/apache/spark/pull/32298/files/2590edf74f78096249b31dffae6ff94f293fc78d#r834760177 

Actually, after https://github.com/apache/spark/pull/34929 the final optimized plan will not contain any `WithCTE` nodes so we can revert this change, but I think the new `scalarSubqueryCTEMultiplicator` param is still needed.